pkgs/applications/misc/orca/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv, lib, pkgconfig, fetchFromGitHub, pkgs
+, autoreconfHook, wrapGAppsHook
+, intltool, yelp-tools, itstool, libxmlxx3
+, python, pygobject3, gtk3
+, at_spi2_atk, at_spi2_core, pyatspi, dbus, dbus-python, pyxdg
+, louis ? null
+, xmodmap, xkbcomp, gsettings_desktop_schemas
+
+, speechdSupport ? true, speechd
+, brailleSupport ? false, brltty, brlapi ? null
+, setproctitleSupport ? true, setproctitle
+, gstreamerSupport ? true, gst_all_1, gst-python
+}:
+
+with lib;
+#pkgs.python3Packages.buildPythonPackage rec {
+stdenv.mkDerivation rec {
+  name = "orca-${version}";
+  version = "3.26.0";
+  # Tags in git are in the form of "ORCA_3_27_3"
+  gittag = concatStrings [ "ORCA_" (stringAsChars (x: if x == "." then "_" else x) version) ];
+
+  src = fetchFromGitHub {
+    owner = "GNOME";
+    repo = "orca";
+    rev = gittag;
+    sha256 = "0xpg7zgkp0kvcfpsfclrm3ssz4ryryfvk7vzbg8d18bdgw0zdw3b";
+  };
+
+  # doCheck = true;
+
+  nativeBuildInputs = [ autoreconfHook wrapGAppsHook pkgconfig libxmlxx3
+    intltool yelp-tools itstool ];
+
+  buildInputs = [ python pygobject3 gtk3 pyatspi dbus-python xkbcomp
+    at_spi2_atk at_spi2_core dbus pyxdg gsettings_desktop_schemas ]
+    ++ optional (speechdSupport) speechd
+    ++ optionals (brailleSupport) [ brltty ]
+    ++ optional (setproctitleSupport) setproctitle
+    ++ optionals (gstreamerSupport) [ gst-python gst_all_1.gstreamer gst_all_1.gst-plugins-base gst_all_1.gst-plugins-bad ]
+  ;
+
+  # Run intltoolize to create po/Makefile.in.in
+  preConfigure = ''
+    intltoolize
+  '';
+#    export GSTREAMER_LIBS=$GST_PLUGIN_SYSTEM_PATH_1_0
+#    export GSTREAMER_CFLAGS=${gst_all_1.gstreamer}/include
+#  '';
+
+  wrapPrefixVariables = [ "PYTHONPATH" "GST_PLUGIN_SYSTEM_PATH_1_0" ];
+
+  preInstall = ''
```

Comment:
Substitution like these are best done in the `postPatch` phase so they are available in all the subsequent phases (especially `checkPhase` which is before `installPhase`).

Reply:
Done, thanks.